addons/mail/controllers/webclient.py

```diff
@@ -10,12 +10,14 @@
 class WebclientController(http.Controller):
     @http.route("/mail/init_messaging", methods=["POST"], type="json", auth="public")
     @add_guest_to_context
-    def mail_init_messaging(self):
+    def mail_init_messaging(self, context=None):
+        if not context:
+            context = {}
         if not request.env.user._is_public():
-            return request.env.user.sudo(False)._init_messaging()
+            return request.env.user.sudo(False).with_context(**context)._init_messaging()
```

Comment:
context forwarding is something we want to especially avoid, this should be changed to pass an actual param with a meaning to the method

Reply:
Noted, I'll create a PR to rectify that as soon as possible.